src/kvstore/raftex/RaftPart.cpp

```diff
@@ -1313,6 +1334,50 @@ cpp2::ErrorCode RaftPart::verifyLeader(
     return cpp2::ErrorCode::SUCCEEDED;
 }
 
+void RaftPart::processSendSnapshotRequest(const cpp2::SendSnapshotRequest& req,
+                                          cpp2::SendSnapshotResponse& resp) {
+    LOG(INFO) << idStr_ << "Receive snapshot, total rows " << req.get_rows().size()
+              << ", total count received " << req.get_total_count()
+              << ", total size received " << req.get_total_size()
+              << ", finished " << req.get_done();
+    std::lock_guard<std::mutex> g(raftLock_);
+    // Check status
+    if (UNLIKELY(status_ == Status::STOPPED)) {
+        VLOG(2) << idStr_
+                << "The part has been stopped, skip the request";
+        resp.set_error_code(cpp2::ErrorCode::E_BAD_STATE);
+        return;
+    }
+    if (UNLIKELY(status_ == Status::STARTING)) {
+        VLOG(2) << idStr_ << "The partition is still starting";
+        resp.set_error_code(cpp2::ErrorCode::E_NOT_READY);
+        return;
+    }
+    if (status_ != Status::WAITING_SNAPSHOT) {
+        LOG(INFO) << idStr_ << "Begin to receive the snapshot";
+        status_ = Status::WAITING_SNAPSHOT;
+        wal_->reset();
+    }
+    // TODO(heng): Maybe we should save them into one sst firstly?
+    // And before commit the snapshot, we should clean up all data.
+    auto ret = commitSnapshot(req.get_rows(),
```

Comment:
Before installing snapshot, We also need to check the following  three  conditions:

1. Whether it is stale term(req.term and current_term).
2. Whether more than one leader (req.leaderId and leader_id).
3. If the current node's leader is not set at this time, then we need to set the leader.

Reply:
Actually,  i have considered this problem.  For snapshot,   we don't care about it is from leader or follower,  because the snapshot is the same.

Reply:
Not the same,  but i still add some checks here to make the logic more simple